.github/workflows/nightly-demo.yaml

```diff
@@ -3,6 +3,9 @@ name: Deploy Nightly Demo
 on:
   workflow_dispatch:
     # Allows for manual triggering.
+  push:
+    branches:
+      - main
```

Comment:
I think we need to double-check that the group ID is correct for this scenario, so that we know cancel-in-progress works.

If github.ref is `main`, we're good.  If it's a SHA1, then we'll have concurrent pushes to the nightly, which will be a problem.

I can never remember how those github variables work.

Also, with this change, do we still need the schedule?

Reply:
Found the right detail in the docs:

> The fully-formed ref of the branch or tag that triggered the workflow run. For workflows triggered by push, this is the branch or tag ref that was pushed.

So that should be "main".